src/papers/infra/arxiv_client.py

```
@@ -0,0 +1,94 @@
+            max_results=limit,
+            sort_by=arxiv.SortCriterion.Relevance,
+        )
+
+        for entry in search.results():  # type: ignore[attr-defined]
+            arxiv_id = getattr(entry, "entry_id", "").rsplit("/", 1)[-1]
+            title = getattr(entry, "title", "").strip()
+            abstract = getattr(entry, "summary", "").strip()
+            authors = [a.name for a in getattr(entry, "authors", [])]
+            pdf_url = None
+            try:
+                pdf_url = next((l.href for l in entry.links if l.title == "pdf"), None)  # type: ignore[attr-defined]
```

Ruff (0.14.2)

Ambiguous variable name: `l`

(E741)